intercom/read_test.go

```diff
@@ -0,0 +1,286 @@
+package intercom
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/interpreter"
+	"github.com/amp-labs/connectors/common/jsonquery"
+	"github.com/amp-labs/connectors/test/utils/mockutils"
+	"github.com/go-test/deep"
+)
+
+func TestRead(t *testing.T) { //nolint:funlen,gocognit,cyclop,maintidx
+	t.Parallel()
+
+	responseErrorFormat := mockutils.DataFromFile(t, "page-req-too-large.json")
+	responseContactsFirstPage := mockutils.DataFromFile(t, "read-contacts-1-first-page.json")
+	responseContactsSecondPage := mockutils.DataFromFile(t, "read-contacts-2-second-page.json")
+	responseContactsThirdPage := mockutils.DataFromFile(t, "read-contacts-3-last-page.json")
+	responseReadConversations := mockutils.DataFromFile(t, "read-conversations.json")
+	responseNotesFirstPage := mockutils.DataFromFile(t, "read-notes-1-first-page.json")
+	responseNotesSecondPage := mockutils.DataFromFile(t, "read-notes-2-last-page.json")
```

Comment:
Last page of contacts and notes slightly difffers in page object.
Array is stored under different field names for conversations.